SiameseDataset.py

```diff
@@ -0,0 +1,271 @@
+from __future__ import print_function
+import collections
+import h5py
+import numpy.random as random
+import numpy as np
+
+from CachedDataset2 import CachedDataset2
+from HDFDataset import FeatureSequenceStreamParser, SparseStreamParser, SegmentAlignmentStreamParser
+from Dataset import DatasetSeq
+
+
+
+class SiameseHDFDataset(CachedDataset2):
```

Comment:
I think it would be cleaner (for now) to not have this in an own separate file but to move it to the `HDFDataset.py`.